### ISSUE TITLE
Reindexing from index with doc type to non-typed mapping fails

### ISSUE THREAD

**Issue by hackery** [elastic/elasticsearch#43100]
**Elasticsearch version** 6.8.0:
**Plugins installed**: None
**JVM version** Oracle 1.8.0_201
**OS version** Linux example 3.10.0-693.21.1.el7.x86_64 #1 SMP Wed Mar 7 19:03:37 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

We're working through the process to migrate to 7.x and remove any document types. All indices we have, have (always had) only one type (e.g. "events" or "transactions"). Trying a reindex from a "typed" index to an "untyped" one (specified with the same template with the type level removed and `?include_type_name=false`) gives a mapping conflict:

```
POST /_reindex
{
  "source": {
    "index": "example-2019-06-09"
  },
  "dest": {
    "index": "example-2019-06-09-test"
  }
}

# task API output ...
"cause": {
        "type": "illegal_argument_exception",
        "reason": "Rejecting mapping update to [example-2019-06-09-test] as the final mapping would have more than 1 type: [_doc, events]"
```

It seems this should be disregarding the type of the source and honouring the un-typed template for creating the new index.

@dadoonet [described a solution](https://discuss.elastic.co/t/rejecting-mapping-update-to-test-as-the-final-mapping-would-have-more-than-1-type-type1-doc/110997/8) on the forum, but this situation and the "type" option should be described in [the documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-reindex.html).

Adding `"type": "_doc"` (not `doc` as given) worked for my reindexing.

**Comment by henningandersen** [comment]
@hackery thanks for your interest in elasticsearch. I noticed that you are running version 6.8 but linking to "current" documentation. When I look in documentation for 6.8:

https://www.elastic.co/guide/en/elasticsearch/reference/6.8/docs-reindex.html

the type option seems to be described adequately. Please let us know if there is additional clarification necessary?

**Comment by jtibshirani** [comment]
The 6.8 reindex documentation also looks correct to me. The 'typeless' reindex + index APIs were only added in 7.0 and are not yet available in 6.x. So when the type is omitted for the destination index, reindex still uses the old behavior where the type is taken from documents in the `_source` index. In 7.0, omitting the type is interpreted as a 'typeless' call, so there is no type conflict error.

I noticed in the [removal of types documentation](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/removal-of-types.html) that we forget to specify the destination type when reindexing. This is probably causing confusion, I opened #45118 to address it.

The behavior on 6.x is quite subtle, but I hope with that correction to the documentation, it will be clearer for users what to do. We also have a blog post which goes through the 7.0 upgrade steps in detail: https://www.elastic.co/blog/moving-from-types-to-typeless-apis-in-elasticsearch-7-0
